app/controllers/reviews_controller.rb

```diff
@@ -12,6 +12,12 @@ def index
 
     @reviews_map[nil] = @reviews.where(semester_id: nil).sort_by{|r| r.course.mnemonic_number}
 
+    average_prof_rating = 0
+    @reviews.each do |r|
+      average_prof_rating += r.overall
+    end
+    @avg_prof_rating = (average_prof_rating/@reviews.count).round(2)
```

Comment:
This is controller logic. You should do essentially the same thing, just in the controller, and use `@avg_prof_rating` in the view like you would any other value. Remember -- views should mostly just be for rendering.